### PR TITLE
ticdc 2 pulsar : add pulsar config from sink-uri & pulsar errors

### PR DESCRIPTION
**Pull request by yumchina**
### What problem does this PR solve?

Add pulsar client and producer‘s config
 
Issue Number: ref #9413

### What is changed and how it works?

It works through sinkURI

#### Tests 

 - Unit test
 

##### Will it cause performance regression or break compatibility?

No, there are some configurations that can be used in the future.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No

### Release note 

None

### CODE REVIEW

File: pkg/sink/pulsar/config.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+	// if batch messages are enabled. If set to a non zero value, messages will be queued until this time
+	// interval or until
+	BatchingMaxPublishDelay = "batching-max-publish-delay"
+
+	// SendTimeout producer max send message timeout  (default: 30000ms)
+	SendTimeout = "send-timeout"
+
+	// MessageKey Option.All Keys filled in Pulsar when sending events are used for routing in Pulsar. If it is empty, it will be randomly distributed in each partition.
+	//For one changefeed task.
+	MessageKey = "message-key"
```
Comment (asddongmen) [inline pkg/sink/pulsar/config.go:51]: Should discuss wether it should be a changefeed level config.

File: docs/design/2023-07-04-ticdc-pulsar-sink.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+#### Pulsar Authentication
+
+- Use authentication-token from sink-uri support token to authenticate the pulsar server.
+- Use basic-user-name and basic-password from sink-uri authenticate to the pulsar server.
+- Use token-from-file from sink-uri support token to authenticate the pulsar server.
+
+#### Pulsar Route Rule
+
+- We support route events to different partitions by message-key.
```
Comment (asddongmen) [inline docs/design/2023-07-04-ticdc-pulsar-sink.md:197]: Please correct this description.

File: pkg/sink/pulsar/config.go [hunk 3]
```diff
@@ -0,0 +1,386 @@
+package pulsar
```
Comment (3AceShowHand) [inline pkg/sink/pulsar/config.go:1]: please add copy right file headers

File: pkg/sink/pulsar/config_test.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
+package pulsar
```
Comment (3AceShowHand) [inline pkg/sink/pulsar/config_test.go:1]: please add copy right file headers

**Review by 3AceShowHand — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**